Qase integration and ios test setup

tobi-legan opened the pull request:
# Description

This PR adds Qase.io test management integration for e2e tests, enabling automatic test result reporting with platform-specific test run naming. It also introduces separate WebdriverIO configurations for iOS and Android platforms, making it easier to run and manage platform-specific tests.

Key additions:
- Custom Qase API wrapper (`qase-wrapper.ts`) that provides a `qase()` function for wrapping test cases
- Platform-specific WebdriverIO configs: `wdio.base.conf.ts`, `wdio.ios.conf.ts`, `wdio.android.conf.ts`
- Automatic platform detection and test run naming (e.g., "E2E Test Run [iOS]" or "E2E Test Run [Android]")
- Error message formatting with ANSI code stripping for clean Qase reports
- Accurate test duration tracking in milliseconds
- Updated npm scripts: `test:ios`, `test:android`, `test:all`

## Motivation and Context

This change is required to:
1. **Centralize test management**: Integrate e2e tests with Qase.io for better test tracking, reporting, and analytics across the team
2. **Improve test organization**: The previous setup had a single WebdriverIO config that required manual switching between iOS and Android capabilities, making it difficult to run platform-specific tests and track results separately
3. **Enhance visibility**: Automatic test result reporting provides real-time visibility into test execution status on both platforms
4. **Better debugging**: Formatted error messages in Qase make it faster to identify and fix test failures
5. **CI/CD readiness**: Foundation for automated test reporting in CI/CD pipelines

The problem this solves: Previously, there was no way to automatically track test results in a centralized test management system, and running tests for different platforms required manual configuration changes.

## Related Issue

PR fixes the following issue: N/A (New feature implementation - no existing issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

- [x] This change requires a documentation update

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1212125996130279

tobi-legan committed:
feat: add E2E testing setup with WebdriverIO and Appium

- Introduced E2E testing framework in the `e2e` directory, including configuration files and test scripts.
- Added `HomeOnboardingScreen` page object for managing onboarding screen interactions.
- Implemented initial test for wallet app onboarding, verifying UI elements and welcome messages.
- Updated `.gitignore` to exclude E2E test artifacts and configuration files.
- Modified `app.json` to disable development client for production builds.
- Added `package-lock.json` and `package.json` for E2E testing dependencies.

tobi-legan committed:
feat: add wallet details test data for E2E testing

tobi-legan committed:
refactor: update Ethereum provider URL and clean up comments in get-chains-config.ts

tobi-legan committed:
feat: Add Qase integration with platform-specific test run naming and multi-platform config support

- Add Qase.io test management integration
- Create platform-specific WebdriverIO configs (iOS/Android)
- Implement qase() wrapper for automatic test result reporting
- Add platform detection for test run naming ([iOS]/[Android])
- Support both iOS and Android e2e testing
- Update README with platform-specific test commands

jonathunne commented on e2e/pageObjects/home-onboarding-screen.ts:
```diff
@@ -0,0 +1,52 @@
+    // Cross-platform text selector
+    // For iOS: Use class chain or accessibility ID
+    // For Android: UiSelector works
+    const isIOS = (driver as any).capabilities.platformName === 'iOS';
```
Possible to use a type here instead of `any` ?

tobi-legan committed:
feat: implement E2E testing framework with WebdriverIO and Appium

- Added configuration files for E2E testing in the `test/e2e` directory.
- Created `HomeOnboardingScreen` page object for managing onboarding interactions.
- Developed initial test for wallet app onboarding, verifying UI elements and welcome messages.
- Updated `package.json` and `package-lock.json` to include necessary dependencies for testing.
- Modified `.gitignore` to exclude test artifacts and configuration files.
- Removed legacy E2E testing setup from `e2e` directory.

tobi-legan committed:
Merge appium-e2e-testing-setup: Update folder structure to root project

tobi-legan committed:
chore: remove duplicate wdio.ios.conf.ts from test/e2e

tobi-legan committed:
chore: update dependencies and configuration for E2E testing

- Added `dotenv`, `qase-api-client`, and `strip-ansi` to `package.json` and `package-lock.json`.
- Updated test specifications path in `wdio.base.conf.ts` to point to the new structure.
- Removed legacy `wdio.conf.ts` file.
- Adjusted TypeScript configuration to exclude `node_modules` and `e2e` directories.

tobi-legan commented on e2e/test/specs/onboarding.e2e.ts:
```diff
@@ -0,0 +1,42 @@
+describe('Onboarding Screen', () => {
+  it('TW-1: First launch', qase('TW-1', async () => {
+    // Wait for app to load
+    await driver.pause(3000);
```
this is done

tobi-legan commented on e2e/data/wallet-details-for-test.ts:
this has been removed

tobi-legan commented on test/e2e/utils/qase-wrapper.ts:
```diff
@@ -0,0 +1,277 @@
+ * Provides a qase() function that matches @tetherto/qase-utils API
+ */
+
+import stripAnsi from 'strip-ansi';
```
yeah we need it for our integration to our test management tool , we send test reports to Qase so we can see test runs formthe tests

tobi-legan committed:
fix: remove all 'any' types and fix Android test configuration

- Replace all 'any' types with proper TypeScript types (WebdriverIO.Capabilities, Record<string, unknown>, etc.)
- Fix Android APK path in wdio.android.conf.ts (test/apps/app-release.apk)
- Fix Android test script in package.json to use wdio.android.conf.ts
- Add missing driver import in home-onboarding-screen.ts
- All tests passing on both iOS and Android

tobi-legan committed:
feat: update E2E testing configuration and add platform-specific setups

- Updated `package.json` to reference new test configuration files for E2E testing.
- Added `wdio.conf.ts` for Android E2E testing setup.
- Added `wdio.ios.conf.ts` for iOS E2E testing setup, including specific capabilities for both platforms.

tobi-legan committed:
Merge appium-e2e-testing-setup: Add test/wdio.conf.ts and move wdio.ios.conf.ts to test/

tobi-legan committed:
refactor: update E2E testing scripts and configurations

- Modified `package.json` to point to the correct paths for E2E testing scripts.
- Added `wdio.base.conf.ts` for shared WebdriverIO configuration across platforms.
- Introduced `wdio.android.conf.ts` for Android-specific WebdriverIO settings, including capabilities and app path.

tobi-legan committed:
Merge branch 'main' into qase-integration-and-ios-test-setup
